### PR TITLE
Feature/json formatter

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

First iteration of the Json Formatter Module.

#### TdJsonFormatterComponent

Added <td-json-formatter> component that facilitates formatting and navigation of JavaScript objects. Mimicking chrome console. This iteration contains:
- Module docs and demo.
- Added package.json
- Added README.md
- Tooltip children preview. ( max elements to render: 5)
- Basic `a11y` (enter and tabbing).

Usage:
```html
<td-json-formatter [data]="objData" [levelsOpen]="1"></td-json-formatter>
```
Properties:
image

#### Test Steps

- [x] `ng serve`
- [x] go to http://localhost:4200/#/components/json-formatter
- [x] Use demo and see documentation.
- [x] Try to use element following documentation anywhere in docs app.
- [x] :v: 

##### Screenshots or link to CodePen/Plunker/JSfiddle
image

json-formatter-demo1

### CODE REVIEW

**Comment by richavyas** [comment]
@emoralesb05 - Is this how it is supposed to function? 

covalent-ljson-formatter-error

**Comment by emoralesb05** [comment]
Removed expand/collapse from demo since it was just for show and in some cases they werent synced with the actual levels opened in json-formatter. (not really needed)

**Comment by kyleledbetter** [comment]
pushed a few cosmetic changes:
image

File: src/platform/json-formatter/json-formatter.component.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  }
+
+  hasChildren(): boolean {
+    return this.children.length > 0;
+  }
+
+  /**
+   * Gets parsed value depending on value type.
+   */
+  getValue(value: any): string {
```
Comment (richavyas) [inline src/platform/json-formatter/json-formatter.component.ts]: scope of this method can be `static`

File: src/platform/json-formatter/json-formatter.component.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          .replace(/\{.*\}/, '') + '{…}';
+    }
+    return value;
+  }
+
+  /**
+   * Gets type of object.
+   * returns 'null' if object is null and 'date' if value is object and can be parsed to a date.
+   */
+  getType(object: any): string {
```
Comment (richavyas) [inline src/platform/json-formatter/json-formatter.component.ts]: Same for this method : scope of this method can be `static`

**Pull request merged**